cpp/src/join/join.cu

```diff
@@ -314,21 +314,27 @@ std::unique_ptr<experimental::table> construct_join_output_df(
       auto common_from_right = experimental::detail::gather(right.select(right_common_col),
                                                             complement_indices.second.begin(),
                                                             complement_indices.second.end(),
-                                                            nullify_out_of_bounds);
+                                                            nullify_out_of_bounds,
+                                                            rmm::mr::get_default_resource(),
+                                                            stream);
       auto common_from_left  = experimental::detail::gather(left.select(left_common_col),
                                                            joined_indices.first.begin(),
                                                            joined_indices.first.end(),
-                                                           nullify_out_of_bounds);
+                                                           nullify_out_of_bounds,
+                                                           rmm::mr::get_default_resource(),
+                                                           stream);
       common_table =
-        experimental::concatenate({common_from_right->view(), common_from_left->view()});
+        experimental::concatenate({common_from_right->view(), common_from_left->view()}, mr);
```

Comment:
```suggestion
        experimental::concatenate({common_from_right->view(), common_from_left->view()}, mr, stream);
```

Reply:
I didn't notice this was the `table` overload of `concatenate`, which doesn't currently have a `detail` variant which takes a stream parameter. I'll file a follow-up issue for this.